src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets

```diff
@@ -210,7 +210,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     it requires a host in the output directory to load the app.
     During "publish", all required assets are copied to the publish directory.
     -->
-    <ItemGroup Condition="'$(SelfContained)' == 'true' and '$(RuntimeIdentifier)' != '' and '$(OutputType)' == 'Exe'">
+    <ItemGroup Condition="'$(SelfContained)' == 'true' and '$(RuntimeIdentifier)' != '' and '$(HasRuntimeOutput)' == 'true'">
```

Comment:
My comment was a little vague.  I meant this should either:

- stay '$(OutputType)' == 'Exe'
or
- use $(_IsExecutable)

I didn't mean the MSBuild code should be ` '$(OutputType)' == 'Exe' or $(_IsExecutable)`